File: reto20/reto20.py
    def fix_gift_list(received: list[str], expected: list[str]) -> dict[str, int]:
        received.sort()
        expected.sort()
        missing: dict[str, int] = {}
        extra: dict[str, int] = {}
        set_gifts = set(received + expected)
        for gift in set_gifts:
            received_count = received.count(gift)
            expected_count = expected.count(gift)
            if received_count > expected_count:
                extra[gift] = received_count - expected_count
            elif received_count < expected_count:
                missing[gift] = expected_count - received_count
        return {"missing": missing, "extra": extra}

File: reto20/reto20.ts
function fixGiftList(
  received: string[],
  expected: string[]
): { missing: Record<string, number>; extra: Record<string, number> } {
  received.sort();
  expected.sort();
  const missing: Record<string, number> = {};
  const extra: Record<string, number> = {};
  const setGifts = new Set([...received, ...expected]);
  for (const gift of setGifts) {
    const receivedCount = received.filter((g) => g === gift).length;
    const expectedCount = expected.filter((g) => g === gift).length;

    if (receivedCount > expectedCount)
      extra[gift] = receivedCount - expectedCount;
    else if (receivedCount < expectedCount)
      missing[gift] = expectedCount - receivedCount;
  }
  return {
    missing: missing,
    extra: extra,
  };
}
